client/samples/lib/sample/vcenter/vm/hardware/adapter/sata.rb

```diff
@@ -0,0 +1,93 @@
+# Copyright 2014-2017 VMware, Inc.  All Rights Reserved.
+# SPDX-License-Identifier: MIT
+require 'sample/framework'
+require 'vapi'
+require 'com/vmware/vcenter/vapi'
+require 'sample/vcenter/helper/vm_helper'
+
+class SATAConfiguration < SampleBase
+  HARDWARE_ADAPTER_MODULE = Com::Vmware::Vcenter::Vm::Hardware::Adapter
+  HARDWARE_ADAPTER_SATA_CLASS = HARDWARE_ADAPTER_MODULE::Sata
+  TITLE = 'SATA Configurations of a VM'
+  DESCRIPTION = <<EOL
+  Demonstrates how to configure virtual SATA adapters of a virtual machine.
+
+  Sample Prerequisites:
+  The sample needs an existing VM.
+EOL
+
+  # sample attributes
+  attr_reader :service_manager,:vm_name,:vm_id,:sata_svc,:satas_to_delete
+
+  def initialize
+    super(TITLE, DESCRIPTION, true)
+  end
+
+  def inject_options
+    option_parser.on("-v", "--vm-name NAME","Name of the VM") do |value|
+      options[:vm_name] =  value
+      end 
+  end
+  
+  def check_options
+    @vm_name = check_required_opt(:vm_name,'--vm-name')
+  end
+
+  def setup
+    ssl_options = {}
+    ssl_options[:verify] = :none if insecure
+    ssl_options[:ca_file] = ca_file if ca_file
+    @service_manager = get_service_manager(ssl_options)
+    @sata_svc = HARDWARE_ADAPTER_SATA_CLASS.new(service_manager.vapi_config)
+  end
+
+  def execute
+    @vm_id = VMHelper.get_vm(service_manager.vapi_config, @vm_name)
+    if not vm_id
```

Comment:
In most cases when you see `if not` or `if !`, you should use `unless` instead. In this case, this line would be `unless vm_id`.